core/system/slot.go

```diff
@@ -104,6 +104,15 @@ func (s *AdaptiveSlot) doCheckRule(rule *Rule) (bool, string, float64) {
 			}
 		}
 		return true, "", c
+	case MemUsage:
+		m := CurrentMemoryUsage()
+		if m > rule.LowWaterMark {
+			if rule.Strategy != BBR || !checkBbrSimple() {
```

Comment:
The implementation of func "checkBbrSimple()" has nothing to do with memory usage.